examples/python/run_llama_batched_vllm.py

```diff
@@ -241,6 +248,71 @@ def _pad_to_max(x: List[int], max_len: int) -> List[int]:
     )
 
 
+def _prepare_eval_queries(
+    requests: List[EvalQueryRequest],
+    all_slot_mappings,
+    sliding_window,
+    dev,
+):
+    seq_lens = []
+    query_lens = []
+    input_ids = []
+    slot_mapping = []
+    past_slot_mapping = []
+    positions = []
+    permute_map = []
+
+    query_offset = sum([request.num_past_tokens for request in requests])
+    past_offset = 0
+
+    for request in requests:
+        num_past_tokens = request.num_past_tokens
+        num_queries = len(request.query_token_ids)
+        query_lens.append(num_queries)
+        request_id = request.request_id
+        input_ids += request.query_token_ids
+
+        positions += [num_past_tokens + i for i in range(num_queries)]
+
+        if sliding_window:
+            seq_lens.append(min(num_past_tokens + num_queries, sliding_window))
+            num_past = min(num_past_tokens, sliding_window)
+            past_slot_mapping += all_slot_mappings[request_id][num_past:]
```

Comment:
Should this be `[:num_past]`?

Reply:
yes fixed now

I haven't tested the sliding-window code path yet. Things get complicated especially with parallel sampling + SWA case.

Reply:
Actually above is still not correct when `num_past_tokens > sliding_window`. The correct one should be
```
past_slot_mapping += all_slot_mappings[request_id][num_past_tokens - sliding_window : num_past_tokens]
```

Reply:
Yes I think this one is correct. `past_slot_mapping` has `sliding_window - num_queries` in it and `slot_mapping` has the remaining `num_queries` tokens